examples/financial_document_analysis_with_llamaindex.ipynb

```diff
@@ -0,0 +1,554 @@
+    "You can get started with just a few lines of code and build a retrieval-augmented generation (RAG) system in minutes.\n",
+    "For more advanced users, LlamaIndex offers a rich toolkit for ingesting and indexing your data, modules for retrieval and re-ranking, and composable components for building custom query engines.\n",
+    "\n",
+    "See [full documentation](https://gpt-index.readthedocs.io/en/latest/) for more details.\n",
+    "\n",
+    "### Financial Analysis over 10-K documents\n",
+    "A key part of a financial analyst's job is to extract information and synthesize insight from long financial documents.\n",
+    "A great example is the 10-K form - an annual report required by the U.S. Securities and Exchange Commission (SEC), that gives a comprehensive summary of a company's financial performance.\n",
+    "These documents typically run hundred of pages in length, and contain domain-specific terminology that makes it challenging for a layperson to digest quickly.\n",
+    "\n",
+    "\n",
+    "We showcase how LlamaIndex can support a financial analyst in quickly extracting information and synthesize insights **across multiple documents** with very little coding. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ad72f9c1-e35c-476d-94d8-ca9619d0cc08",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15284a54-35b3-4129-9a7a-57b7547bfaf4",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "To begin, we need to install the llama-index library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1062e24-6177-459a-94e0-8e77b3e0859b",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pip install llama-index"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f51cb5df-9c98-44d8-8c69-1a0b784de3e3",
+   "metadata": {},
+   "source": [
+    "Now, we import all modules used in this tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "09fbec4c-1864-4d76-9dbf-3d213ba58fc8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from langchain import OpenAI\n",
+    "\n",
+    "from llama_index import SimpleDirectoryReader, ServiceContext, VectorStoreIndex\n",
+    "from llama_index import set_global_service_context\n",
+    "from llama_index.response.pprint_utils import pprint_response\n",
+    "from llama_index.tools import QueryEngineTool, ToolMetadata\n",
+    "from llama_index.query_engine import SubQuestionQueryEngine"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c743f504-f28c-4802-89b6-ad152b74b0eb",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Before we start, we can configure the LLM provider and model that will power our RAG system.  \n",
+    "Here, we pick *text-davinci-003* from OpenAI, allow unlimited output tokens.  "
```

Comment:
Is there a particular reason we use the Completions model rather than the Chat models? Given our new ones are Chat-based, it'd be great to refactor this to use them if possible.

If you don't currently support this we can push as-is and make a subsequent notebook.

Reply:
sure, we can swap to chat models. Right now we use langchain LLM abstractions so it's a one-line change.

Reply:
Hey @colin-openai , just tried to update the notebook to using the latest `gpt-3.5-turbo-0613` model, but was getting some qualitatively worse answers (model refusing to give an answer, saying "Based on the provided context information, it is not explicitly mentioned...."). 

I think we might need to do some prompt tuning to update for the new model. Maybe we can just land this as is, and then update this after we do some more experiments?